### ISSUE TITLE
[BUG] - Linux Client Denies Connection

### ISSUE THREAD

**Issue by Rastuasi** [invernyx/smartcars-3-bugs#329]
### Describe the bug

Downloaded the latest plugin and installed as of tonight. Loaded a bid from Walker Air and get two options as the return. In 20 loads, 18 were the screenshot attached, saying cannot connect, the 2 others were "invalid scenario" and it claiming no flight plan, but one is clearly there.

### How do you reproduce this bug?

1. Install latest linux client on Manjaro
2. Load sim and at airport.
3. Load smartcars3, go to bid screen, find the single bid, activate bid
4. Wait for failure.

### Expected behavior

It would connect and track flight

### Screenshots

![image](https://github.com/invernyx/smartcars-3-bugs/assets/17036015/617d1409-4104-4ad1-bb40-65e89be86ed2)
[combined-2023-08-30_bug.log](https://github.com/invernyx/smartcars-3-bugs/files/12481142/combined-2023-08-30_bug.log)
![image](https://github.com/invernyx/smartcars-3-bugs/assets/17036015/be14d455-ca88-4f7e-b0eb-a30fdeea5d67)
![image](https://github.com/invernyx/smartcars-3-bugs/assets/17036015/be5be4b9-6570-4fc6-83e5-b0b7c8cb95a7)

### Operating system

Manjaro Arch Linux

### Community airline

Walker Air

### smartCARS Version

0.14.0

### Plugins installed

chat, flight center, flight tracking, logg

**Comment by Rastuasi** [comment]
Also forgot to mention, I did already try clearing out the .config/smartCARS 3 files as well when updating, so it should not be an old config file from the previous installations.

**Comment by Rastuasi** [comment]
[error-2023-08-30.log](https://github.com/invernyx/smartcars-3-bugs/files/12481234/error-2023-08-30.log)
Just for addition, this file was also there, but the errors are probably from me trying my pilot ID instead of email to log in.

**Comment by GenericNerd** [comment]
This is not a bug. You need to change your settings to use the X-Plane client, rather than the FSUIPC client which is currently selected.

**Comment by Rastuasi** [comment]
> This is not a bug. You need to change your settings to use the X-Plane client, rather than the FSUIPC client which is currently selected.

Reread it and know your software. The linux client defaults to only xplane. I am not selecting the FSUIPC client, that's just what your error says. Please reopen

**Comment by Rastuasi** [comment]
Already provided a screenshot, but just to prove it again: 
![image](https://github.com/invernyx/smartcars-3-bugs/assets/17036015/b187c420-ce40-4db7-8eae-5b0d6494eb4d)
As you can see, NO FSUIPC client option and no option for Xplane, because it DEFAULTS to XPlane, bug is valid, reopen it please and investigate why the last version connected through UDP but the new one fails to connect.

**Comment by GenericNerd** [comment]
You're right, I forgot that we have platform-specific options and the tracker is one of them. I've re-opened the issue.

Can you try setting port 49010 in smartCARS instead of the legacy port 49000?

**Comment by Rastuasi** [comment]
No worries, I did try changing the ports in mix of those 20 attempts and restarts, but didn't save off a log, let me get that real quick.

**Comment by Rastuasi** [comment]
@GenericNerd no luck. I opened smartcars, set the setting, closed it fully, loaded sim up, had wheels on ground, bid placed, plane loaded, started smartcars, confirmed setting held the new port and attempted to connect. Same error. 

One thing I did notice, I don't have the option for setting x-plane in the UI, but the config.json has a setting line there and it does say fsupic. 

So for testing, I tried switching that myself to xplane and tried x-plane. Still no connection on either port, but it did change the error to say "cannot connect to simulator" and no reference to fsuipc. So I am guessing that Linux truly does default and ignores that config, but the error message tricks you to think it's looking for fsuipc.

[combined-2023-08-31_49010_attempt.log](https://github.com/invernyx/smartcars-3-bugs/files/12491506/combined-2023-08-31_49010_attempt.log)
![image](https://github.com/invernyx/smartcars-3-bugs/assets/17036015/d6ba3610-6e75-49c8-bbab-f38c24503754)

**Comment by Rastuasi** [comment]
okay was able to find that it should be "xplane" if this were a windows client, where I could set it, via looking at the:

SmartCars/resources/app/dist/plugins/com.tfdidesign.flight-tracking/plugin.json

Setting that manually then in my .config/smartCARS 3/config.json presents this as the error: 

![image](https://github.com/invernyx/smartcars-3-bugs/assets/17036015/77352397-3d3b-4b88-a0cb-a60c72770343)

So removes the reference of FSUIPC, but still no connection on either port.

**Comment by Thien42** [comment]
The FSUIPC error message is only a visual bug which will be fixed, as we use the X-Plane tracker on any non-Windows environment regardless of the `tracking_provider` value.

With that being said, I was not able to reproduce the aforementioned issue. Could it be that you have a firewall that is blocking incoming traffic on either the X-Plane port (49000) or the port smartCARS 3 uses (7172)?

**Comment by Rastuasi** [comment]
I have other apps that use UDP of XP, they don't have issues, so I know 49000 works. Did smartcars3 change to 7172 from something else? Up until this latest version I didn't have issues, so long as I loaded sim first. I'd get this error before if I loaded SC3 before the sim, but it's consistent now, regardless of order

**Comment by Thien42** [comment]
Yes, before being 7172, it was a random port assigned by the OS but that would cause other issues in the long run.

I will add an option to manually select a port, hopefully that will fix the issue for you

**Comment by Rastuasi** [comment]
Great! Well outside my router having a firewall externally, the internal sim communication should be firewall-less, but if the sim connection requires both ends, could be an issue. I'll do some digging on this new port you mentioned and then report back, at least as a way to verify or whatnot on if this is the issue or not

**Comment by Rastuasi** [comment]
Saw that there was a new client, loaded it in, did port forwarding on 7172 on router, tried setting XP to do its own UPnP, tried 49000 and 49010, all with the same results.
[combined-2023-09-01_latest_PortForwarding.log](https://github.com/invernyx/smartcars-3-bugs/files/12501964/combined-2023-09-01_latest_PortForwarding.log)
[error-2023-09-01.log](https://github.com/invernyx/smartcars-3-bugs/files/12501965/error-2023-09-01.log)

**Comment by Thien42** [comment]
I'm not sure what you mean, there is no need for either port forwarding or UPnP to track a flight with smartCARS 3. The port to be used for X-Plane is 49000

**Comment by Rastuasi** [comment]
Not meaning much than forcing the ports that worked prior to be listed on the network, that's all, but even that didn't work. It really shouldn't be a marker though on my network, since the previous version worked. So essentially, I am just trying things and reporting what I am trying so that it may point your dev team into what might be my issue, either way.

**Comment by Rastuasi** [comment]
Additional follow up, I reverted all my testing changes, back to what router was before changes suggested in this ticket on ports and had a flight tracked. Not sure what or why, but it picked up the 7172 just fine, worked like it did before 0.14.0. Not sure why the past several days was a no go though. Going to do a few more test flights and report back.

Does the UDP to XP go out and away from the home network to communicate with SC3? I would not think it would, same network, same machine, it should just connect, wondering if something external was the issue. Just weird how the night before the update, I did a flight, updated and since then, had issues.
